src/Routers/FunctionsRouter.js

```diff
@@ -51,80 +54,58 @@ export class FunctionsRouter extends PromiseRouter {
     });
   }
 
-  static handleCloudJob(req) {
+  static async handleCloudJob(req) {
     const jobName = req.params.jobName || req.body.jobName;
     const applicationId = req.config.applicationId;
     const jobHandler = jobStatusHandler(req.config);
-    const jobFunction = triggers.getJob(jobName, applicationId);
+    const jobFunction = getJob(jobName, applicationId);
     if (!jobFunction) {
       throw new Parse.Error(Parse.Error.SCRIPT_FAILED, 'Invalid job.');
     }
-    let params = Object.assign({}, req.body, req.query);
-    params = parseParams(params);
+    const params = parseParams({ ...req.body, ...req.query });
     const request = {
-      params: params,
+      params,
       log: req.config.loggerController,
       headers: req.config.headers,
       ip: req.config.ip,
       jobName,
       message: jobHandler.setMessage.bind(jobHandler),
     };
 
-    return jobHandler.setRunning(jobName, params).then(jobStatus => {
-      request.jobId = jobStatus.objectId;
-      // run the function async
-      process.nextTick(() => {
-        Promise.resolve()
-          .then(() => {
-            return jobFunction(request);
-          })
-          .then(
-            result => {
-              jobHandler.setSucceeded(result);
-            },
-            error => {
-              jobHandler.setFailed(error);
-            }
-          );
-      });
-      return {
-        headers: {
-          'X-Parse-Job-Status-Id': jobStatus.objectId,
-        },
-        response: {},
-      };
+    const jobStatus = await jobHandler.setRunning(jobName, params);
+    request.jobId = jobStatus.objectId;
+    // run the function async
+    process.nextTick(() => {
+      (async () => {
+        try {
+          const result = await jobFunction(request);
```

Comment:
Plus not sure if this is how jobs should work

Reply:
What do you mean

Reply:
Just wasn't sure if this would run in background like the current jobs do

Reply:
Looks good, your return 200 OK immediately and start executing the job after the current event loop.

Reply:
worst case, it doesn't work and we realize that we're missing a test case